crates/state/src/tx.rs

```diff
@@ -17,7 +17,33 @@ pub enum ProtocolOperation {
     DepositRequest(DepositRequestInfo),
     /// Checkpoint data
     Checkpoint(SignedBatchCheckpoint),
-    // TODO: add other kinds like Proofs and statediffs
+    // TODO: add other kinds like proofs and state diffs
+}
+
+/// Similar to [`ProtocolOperation`] except that this also contains blob data which is not relevant
+/// to chain.
```

Comment:
I don't see DA blobs here.

Also it *is* relevant, we need to see the blob commitments.

I'm a bit confused about what the purpose of this type is, being distinct from the other one.

Reply:
> Also it is relevant, we need to see the blob commitments.

Yes, the commitments, but not the actual data/blob which is what I ment in the docstring.

Reply:
> 'm a bit confused about what the purpose of this type is, being distinct from the other one.

So the idea here is that `RawProtocolOp` would contain a variant `Da(Vec<u8>)` whereas `ProtocolOp` would have the counterpart `DaCommitment(Buf32)`. The other variants would be the same. Those are not here right now because we don't actually have DA at the moment.

Reply:
Just do the visitor thing.  The approach with the handler function is just an roundabout and inelegant way of doing the same general idea.

Reply:
PTAL